docs/dataset.md

```diff
@@ -0,0 +1,45 @@
+# The `BatteryDataset` Object
```

Comment:
I just want to say I think this is a fantastic idea! A code-free full description of the main class we use will make it much easier for us to use and alter it as needs arise, as well as for other users to smoothly get started. Nicely done!

Reply:
Yep, we totally need this for a bunch of features. Will gradually piece it together